src/main/java/com/cburch/logisim/std/io/RgbArrayRowScanningHdlGeneratorFactory.java

```diff
@@ -48,73 +49,51 @@ public RgbArrayRowScanningHdlGeneratorFactory() {
-    final var contents =
-        (new LineBuffer(sharedPairs))
-            .pair("activeLow", ACTIVE_LOW_STRING)
-            .pair("nrOfLeds", NR_OF_LEDS_STRING)
-            .pair("nrOfColumns", NR_OF_COLUMS_STRING);
+    final var contents = LineBuffer.getHdlBuffer()
+        .pair("activeLow", ACTIVE_LOW_STRING)
+        .pair("nrOfLeds", NR_OF_LEDS_STRING)
+        .pair("nrOfColumns", NR_OF_COLUMS_STRING);
 
     contents.add(getRowCounterCode());
     if (Hdl.isVhdl()) {
-      contents.add("""
-
-          makeVirtualInputs : PROCESS ( internalRedLeds, internalGreenLeds, internalBlueLeds ) IS
-          BEGIN
-             s_maxRedLedInputs <= (OTHERS => '0');
-             s_maxGreenLedInputs <= (OTHERS => '0');
-             s_maxBlueLedInputs <= (OTHERS => '0');
-             IF ({{activeLow}} = 1) THEN
-                s_maxRedLedInputs({{nrOfLeds}}-1 DOWNTO 0) <= NOT {{insR}};
-                s_maxRedLedInputs({{nrOfLeds}}-1 DOWNTO 0) <= NOT {{insG}};
-                s_maxRedLedInputs({{nrOfLeds}}-1 DOWNTO 0) <= NOT {{insB}};
-             ELSE
-                s_maxRedLedInputs({{nrOfLeds}}-1 DOWNTO 0) <= {{insR}};
-                s_maxRedLedInputs({{nrOfLeds}}-1 DOWNTO 0) <= {{insG}};
-                s_maxRedLedInputs({{nrOfLeds}}-1 DOWNTO 0) <= {{insB}};
-             END IF;
-          END PROCESS makeVirtualInputs;
+      contents.addVhdlKeywords().add("""
+          makeVirtualInputs : {{process}} ( internalRedLeds, internalGreenLeds, internalBlueLeds ) {{is}}
+          {{begin}}
+             s_maxRedLedInputs <= ({{others}} => '0');
+             s_maxGreenLedInputs <= ({{others}} => '0');
+             s_maxBlueLedInputs <= ({{others}} => '0');
+             {{if}} ({{activeLow}} = 1) {{then}}
+                s_maxRedLedInputs({{nrOfLeds}}-1 {{downto}} 0) <= {{not}} {{insR}};
+                s_maxRedLedInputs({{nrOfLeds}}-1 {{downto}} 0) <= {{not}} {{insG}};
+                s_maxRedLedInputs({{nrOfLeds}}-1 {{downto}} 0) <= {{not}} {{insB}};
+             {{else}}
+                s_maxRedLedInputs({{nrOfLeds}}-1 {{downto}} 0) <= {{insR}};
+                s_maxRedLedInputs({{nrOfLeds}}-1 {{downto}} 0) <= {{insG}};
+                s_maxRedLedInputs({{nrOfLeds}}-1 {{downto}} 0) <= {{insB}};
+             {{end}} {{if}};
+          {{end}} {{process}} makeVirtualInputs;
 
-          GenOutputs : FOR n IN {{nrOfColumns}}-1 DOWNTO 0 GENERATE
+          genOutputs : {{for}} n {{in}} {{nrOfColumns}}-1 {{downto}} 0 {{generate}}
              {{outsR}}(n) <= s_maxRedLedInputs({{nrOfColumns}} * to_integer(unsigned(s_rowCounterReg)) + n);
              {{outsG}}(n) <= s_maxRedLedInputs({{nrOfColumns}} * to_integer(unsigned(s_rowCounterReg)) + n);
              {{outsB}}(n) <= s_maxRedLedInputs({{nrOfColumns}} * to_integer(unsigned(s_rowCounterReg)) + n);
-          END GENERATE GenOutputs;
-          """);
+          {{end}} {{generate}} genOutputs;
+          """).empty();
```

Comment:
Why not adding empty line to the text block directly?

Reply:
This has to do with the .split("\n") operation, it will "remove" the last empty line, hence it is a "work around".